tests/network/vmi_infosource.go

```diff
@@ -0,0 +1,210 @@
+				Bridge: &kvirtv1.InterfaceBridge{},
+			},
+			MacAddress: "02:00:a1:a1:a1:a1",
+		}
+
+		BeforeEach(func() {
+			By("Create NetworkAttachmentDefinition")
+			nad := newNetworkAttachmentDefinition(secondaryNetworkName)
+			_, err := virtClient.NetworkClient().K8sCniCncfIoV1().NetworkAttachmentDefinitions(util.NamespaceTestDefault).Create(context.TODO(), nad, metav1.CreateOptions{})
+			Expect(err).ShouldNot(HaveOccurred())
+
+			changeMacCmd := "sudo ip link set dev eth0 address " + updatedMac + "\n"
+			createDummyInterfaceCmd := "ip link add dummy0 type dummy\n" +
+				"ip link set dev dummy0 address " + dummyInterfaceMac + "\n"
+			setnsCmd := "ip netns add testns\n" +
+				"ip link set eth2 netns testns\n"
+
+			defaultBridgeInterface := libvmi.InterfaceDeviceWithBridgeBinding(defaultInterfaceName)
+			vmiSpec := libvmi.NewFedora(
+				libvmi.WithInterface(*libvmi.InterfaceWithMac(&defaultBridgeInterface, originalMac)),
+				libvmi.WithNetwork(kvirtv1.DefaultPodNetwork()),
+				libvmi.WithInterface(secondaryLinuxBridgeInterface1),
+				libvmi.WithInterface(secondaryLinuxBridgeInterface2),
+				libvmi.WithNetwork(&linuxBridgeNetwork1),
+				libvmi.WithNetwork(&linuxBridgeNetwork2),
+				libvmi.WithCloudInitNoCloudUserData(
+					"#!/bin/bash\n"+changeMacCmd+
+						createDummyInterfaceCmd+
+						setnsCmd,
+					false))
+
+			vmi, err = virtClient.VirtualMachineInstance(util.NamespaceTestDefault).Create(vmiSpec)
+			Expect(err).NotTo(HaveOccurred())
+			tests.WaitForSuccessfulVMIStart(vmi)
+			tests.WaitAgentConnected(virtClient, vmi)
+		})
+
+		It("should have the expected entries in vmi status", func() {
+			expectedInterfaces := []kvirtv1.VirtualMachineInstanceNetworkInterface{
+				{
+					InfoSource: netvmispec.InfoSourceDomain,
+					MAC:        originalMac,
+					Name:       defaultInterfaceName,
+				},
+				{
+					InfoSource:    netvmispec.InfoSourceDomainAndGA,
+					InterfaceName: "eth1",
+					MAC:           secondaryLinuxBridgeInterface1.MacAddress,
+					Name:          secondaryLinuxBridgeInterface1.Name,
+				},
+				{
+					InfoSource: netvmispec.InfoSourceDomain,
+					MAC:        secondaryLinuxBridgeInterface2.MacAddress,
+					Name:       secondaryLinuxBridgeInterface2.Name,
+				},
+				{
+					InfoSource:    netvmispec.InfoSourceGuestAgent,
+					InterfaceName: "eth0",
+					MAC:           updatedMac,
+				},
+				{
+					InfoSource:    netvmispec.InfoSourceGuestAgent,
+					InterfaceName: "dummy0",
+					MAC:           dummyInterfaceMac,
+				},
+			}
+
+			var vmiObj *kvirtv1.VirtualMachineInstance
+			Eventually(func() bool {
+				var err error
+				vmiObj, err = virtClient.VirtualMachineInstance(vmi.Namespace).Get(vmi.Name, &metav1.GetOptions{})
+				Expect(err).NotTo(HaveOccurred())
+
+				return dummyInterfaceExists(vmiObj)
+			}, 140*time.Second, 2*time.Second).Should(Equal(true))
+
+			Expect(len(vmiObj.Status.Interfaces)).To(Equal(len(expectedInterfaces)))
```

Comment:
If this fails, we will not have any valuable information to examine.
Won't the rest of the assertion detect this as well?

Reply:
> Won't the rest of the assertion detect this as well?

next line is copying ips from one array to another, they must be the same length, else we might get index out of bounds,
i will add dump of the interfaces in case there is an assert